homeassistant/components/sensor/deutsche_bahn.py

```diff
@@ -67,15 +67,17 @@ def state(self):
     def device_state_attributes(self):
         """Return the state attributes."""
         connections = self.data.connections[0]
-        connections['next'] = self.data.connections[1]['departure']
-        connections['next_on'] = self.data.connections[2]['departure']
+        if len(self.data.connections) > 1:
+            connections['next'] = self.data.connections[1]['departure']
+        elif len(self.data.connections) > 2:
+            connections['next_on'] = self.data.connections[2]['departure']
```

Comment:
This does not make sense.
Are you sure it should be `elif` here and not only `if`?

Reply:
You're absolutely right. fixed. Thx